### ISSUE TITLE
Is this a new version of bac-002-wifi?

### ISSUE THREAD

**Issue by kobi-wan** [fashberg/WThermostatBeca#124]
Hello,
I ordered a set of new GALWW thermostats from Moes. After I opened the device, I saw a different  label than I suspected. It is labeled "ZWX-GA/GC-CONTROL" and it looks like it's from October 2020. Also the layout is a little bit different. The ESP seems the same (TYWE3S).
So my question is: does it work with ThermostatBecaWifi? Has anybody already flashed the firmware on these version? I don't know if I would try it, because I could still send it back.

![PXL_20210606_105339605](https://user-images.githubusercontent.com/28985724/120922226-2d422300-c6c8-11eb-8d73-d0b6e2995ef7.jpg)

**Comment by tsrubar** [comment]
Yes, these are without backlight of central button, back side ribbon has different number of wires, but they flash just fine. I have done 7 of these 2 weeks ago. Imho enclosure looks and feels much cheaper, what is unfortunate.

**Comment by kobi-wan** [comment]
Okay, thank you. I already noticed, that the central button has no light anymore. So it's better to get one of the older designs imho.

**Comment by maierei** [comment]
Hello @tsrubar,

I also got the same device version but was not able flash the new firmware from Klaus Ahrenberg. I soldered the wires just like in his manual, but cant get any connection using a USB programmer. 
Do you have any hints for me maybe?
Which exact wiring did you use? 
Did you connect the  ESP-GPIO0 to GND of the programmer when powering the ESP?
Did you connect ESP-GPIO15 to GND of the programmer or is this not needed?

Which programmer did you use?

Thanks
Markus

**Comment by tsrubar** [comment]
Hi,
it is kind of hard to remember right now, but iirc I connected 3.3V from programmer to Vcc and En on thermostat, Gnd from programmer to Gnd and GPIO0 on thermostat (to boot in flash mode) and Rx/TX on programmer to Tx/Rx on thermostat.
As a programmer I used cheap NodeMCU.

**Comment by maierei** [comment]
@tsrubar 
Hi, thanks a lot - That's how I tried it - but I used a different programmer and also read that it coul be some kind of power drain - so I have ordered the same programming unit and will test it with that one.
Thanks for gettng back so quick and looking for your post.

**Comment by maierei** [comment]
@tsrubar 
My new programmer arrived today, but it's sad it does not work.
I always get 
A fatal error occurred: Failed to connect to Espressif device: Timed out waiting for packet header
It looks like you have connected an additional BLUE cable somewhere on the board - what's that?

**Comment by tsrubar** [comment]
> @tsrubar My new programmer arrived today, but it's sad it does not work. I always get A fatal error occurred: Failed to connect to Espressif device: Timed out waiting for packet header It looks like you have connected an additional BLUE cable somewhere on the board - what's that?

Oh, I am sorry to hear that. That blue cable is just GND from programmer connected to more accessible place on PCB of thermostat directly connected with GND on TYWE3S. I was trying hard to find the programmer, I used, but we've just moved in our new house and everything is just a mess packed in boxes.
If you have it connected as described, did you try to push reset button few times? Iirc, I had to do it from time to time, but in the end, i managed to flash all 9 of my thermostats. Could you take photo of your setup, so maybe I could possibly notice something that you could potentionally have missed?

**Comment by maierei** [comment]
I just thought I got it working as I used 5V and hit GND with RST and GPIO0 a few times because suddenly I got a connection, but:
- Reading and erasing the flas was not supported
- Writing was supported and it took about 10 times until the counter run to 100% (the tests before stoped at 10 to 20%).

Now I have reassembled the device and hoped I get an Wifi from it to configere but it seems just nothing happened...

Any thougts?

**Comment by tsrubar** [comment]
@maierei dont do it with 5V, you could burn it. Flashing procedure do erasing first and then writing. I was using tasmotizer to flash tasmota to it. You could possibly try to plug it in stronger usb port.
Just found it!
![IMG_20220316_223835](https://user-images.githubusercontent.com/31758598/158695713-1417878a-c503-407d-9751-a2d97a217440.jpg)

**Comment by maierei** [comment]
@tsrubar - ok I will give it a try tomorrow when my girls look at "Germanys next Topmodel"...
So you used Tasmotizer and the firmware from here: https://github.com/klausahrenberg/WThermostatBeca ?
I will try Tasmotizer tomorrow.
Thanks
Markus

**Comment by tsrubar** [comment]
@maierei I cant remember where I got if from, but as far as md5 check for downloaded file is ok, it doesnt really matter where will you download from :) What I exactlly used was: tasmotizer-1.2.exe, but its over 38mb in size, so i temp-hosted it here: https://www.file.io/Gtyi/download/0T8UfQa199gl

**Comment by PatrickHuetter** [comment]
Did you get it to work? I have the same problem with newly arrived BAC-002 Wifi thermostats. I already tested 2 different programmers and i'm sure that i soldered everything correctly. Which programmer did you use?

**Comment by kobi-wan** [comment]
@PatrickHuetter: I can't tell, because I exchanged it with the older one.

**Comment by PatrickHuetter** [comment]
Thanks @kobi-wan . I ordered a [different programming device](https://amzn.to/3BTkHHo). Optically the same one that @tsrubar has used on his photos. Hopefully this will work since i have many thermostats of this model that i want to flash. I'm currently waiting for it to be delivered. I'll report wether it worked or not.

**Comment by PatrickHuetter** [comment]
@maierei Did you get it to work? I have 3 different programmer devices now and none of them worked. I also opened another thermostat (bought many of them) and soldered again and tested with different computers. I can't get a connection to the ESP.

```
esptool.py -p /dev/cu.usbserial-10 -b 460800 read_flash 0x00000 0x100000 originalFirmware1M.bin                                                                                                       ✔
esptool.py v4.2
Serial port /dev/cu.usbserial-10
Connecting......................................

A fatal error occurred: Failed to connect to Espressif device: No serial data received.
For troubleshooting steps visit: https://docs.espressif.com/projects/esptool/en/latest/troubleshooting.html
```

Does somebody have another idea what to do now? Maybe it's the same problem like described here:
https://github.com/fashberg/WThermostatBeca/issues/113#issuecomment-810168238
https://github.com/fashberg/WThermostatBeca/issues/113#issuecomment-820177953
https://github.com/fashberg/WThermostatBeca/issues/115#issuecomment-818619946

![IMG_0707](https://user-images.githubusercontent.com/1142911/184252336-d636520a-b4c7-439a-b51a-049ab61ca299.jpeg)

**Comment by fashberg** [comment]
Hey!
Compared to 2019 Version some changes - see image below

Check my post here: https://github.com/fashberg/WThermostatBeca/issues/115#issuecomment-1264667961

The 5V MCU to 3,3V ESP conversion seems still to be there, so if the MCU is still at 5 Volt and not powered, because you only connect the 3.3volt lane then probably it's something else. Either not enough 3.3 Volt power or some kind of hardware flashing protection.

Where have you bought the new version?

![compare](https://user-images.githubusercontent.com/3075214/193463179-be0ecb99-80c3-4361-b08f-17e9e53994c0.jpg)

**Comment by PatrickHuetter** [comment]
I bought 7 pieces of them on aliexpress. I already tested 4 different programmers and today i tested with a 3d printed programming jig (the one from the flashing.md). Always the same problem. 

```
esptool.py --port /dev/cu.usbserial-A50285BI -b 460800 flash_id                                                                                         2 ✘  8s 
esptool.py v4.3
Serial port /dev/cu.usbserial-A50285BI
Connecting......................................

A fatal error occurred: Failed to connect to Espressif device: No serial data received.
For troubleshooting steps visit: https://docs.espressif.com/projects/esptool/en/latest/troubleshooting.html
```

I dont't know what to do now. Maybe i have to desolder them or just buy the zigbee version and sell the wifi ones.

**Comment by rlehfeld** [comment]
I brought 4 Moes BHT-002-GALW(WiFi) and the PCB looks like in [picture](https://user-images.githubusercontent.com/31758598/120930338-2a0d5e00-c6ed-11eb-8bdc-608e6abac4f0.jpg) from [tsrubar](https://github.com/tsrubar) apart from the fact that the MCU does not have any kind of labeling in none of the 4 devices. It also says ZWX-GA/GC-CONTROL 2020.10.17 First I also faced problems in flashing the device. I tried several things. What actually helped at the end was connecting the position in the middle of J2 with GND. This one is connected to one of the pins of the MCU. I believe it is either an enable or rst pin of the MCU. At least it worked for me. It might be helpfull for others as well.

So in total I had following cabling:

3V3 <----> VCC
3V3 <----> EN
GND <----> GND
GND <----> GPIO0
TX <----> RXD0 
RX <----> TXD0 
middle J2 <---> GND

**Comment by Sub-7** [comment]
Successfully flashed four thermostats as in the first picture.
Wiring was as follows:
![TYWES3S](https://user-images.githubusercontent.com/30667941/209029571-7788d619-1694-41f0-8c33-459a68a14d22.png)

**Comment by Kolia56** [comment]
Hello to everyone. I've been trying to flash an ME81H thermostat with exactly the same wiring as above. It is referenced at [bladadder](https://templates.blakadder.com/ME81H.html ) It features a WBR3 module instead of the TYWE3S. Pinout is the same. I'm using a CH340G as suggested by Tasmota. I've successfully flashed a Shelly 2.5 using this adapter. 
I tried many things before posting. I even tried to ground the RST pin of the [MCU](https://datasheet.lcsc.com/lcsc/1912111437_SOC-Shenzhen-SinOne-Microelectronics-SC92F8547P48R_C367200.pdf), still no success.
To start with I've been using epstool flash_id that returns connection error.
Does anyone have an idea how I could progress?

@PatrickHuetter: did you buy the programming jig at some place?

Thank you
